grp/basic.gd

```diff
@@ -343,64 +343,122 @@ end );
 
 #############################################################################
 ##
-#O  QuaternionGroupCons( <filter>, <n> )
+#O  DicyclicGroupCons( <filter>, <n> )
 ##
 ##  <ManSection>
-##  <Oper Name="QuaternionGroupCons" Arg='filter, n'/>
+##  <Oper Name="DicyclicGroupCons" Arg='filter, n'/>
 ##
 ##  <Description>
 ##  </Description>
 ##  </ManSection>
 ##
-DeclareConstructor( "QuaternionGroupCons", [ IsGroup, IsInt ] );
+DeclareConstructor( "DicyclicGroupCons", [ IsGroup, IsInt ] );
+DeclareSynonym("QuaternionGroupCons", DicyclicGroupCons);
 
 
 #############################################################################
 ##
-#F  QuaternionGroup( [<filt>, ]<n> )  . . . . . . . quaternion group of order <n>
+#F  DicyclicGroup( [<filt>, [<field>, ] ]<n> )  Dicyclic group of order <n>
 ##
-##  <#GAPDoc Label="QuaternionGroup">
+##  <#GAPDoc Label="DicyclicGroup">
 ##  <ManSection>
-##  <Func Name="QuaternionGroup" Arg='[filt, ]n'/>
-##  <Func Name="DicyclicGroup" Arg='[filt, ]n'/>
+##  <Func Name="DicyclicGroup" Arg='[filt, [field, ]]n'/>
+##  <Func Name="QuaternionGroup" Arg='[filt, [field, ]]n'/>
 ##
 ##  <Description>
-##  constructs the generalized quaternion group (or dicyclic group) of size
-##  <A>n</A> in the category given by the filter <A>filt</A>.  Here, <A>n</A>
-##  is a multiple of 4.
+##  <Ref Func="DicyclicGroup"/> constructs the dicyclic group of size <A>n</A>
+##  in the category given by the filter <A>filt</A>. Here, <A>n</A> must be a
+##  multiple of 4. The synonym <Ref Func="QuaternionGroup"/> for
+##  <Ref Func="DicyclicGroup" /> is provided for backward compatibility, but will
+##  print a warning if <A>n</A> is not a power of <M>2</M>.
 ##  If <A>filt</A> is not given it defaults to <Ref Func="IsPcGroup"/>.
 ##  For more information on possible values of <A>filt</A> see section
 ##  (<Ref Sect="Basic Groups"/>).
 ##  Methods are also available for permutation and matrix groups (of minimal
 ##  degree and minimal dimension in coprime characteristic).
 ##  <P/>
 ##  <Example><![CDATA[
-##  gap> QuaternionGroup(32);
-##  <pc group of size 32 with 5 generators>
+##  gap> DicyclicGroup(24);
+##  <pc group of size 24 with 4 generators>
 ##  gap> g:=QuaternionGroup(IsMatrixGroup,CF(16),32);
 ##  Group([ [ [ 0, 1 ], [ -1, 0 ] ], [ [ E(16), 0 ], [ 0, -E(16)^7 ] ] ])
 ##  ]]></Example>
 ##  </Description>
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-BindGlobal( "QuaternionGroup", function ( arg )
+BindGlobal( "GRPLIB_DicyclicParameterCheck",
+function(args, quaternion)
+    local size;
+
+    if Length(args) = 1 then
+        res := [ IsPcGroup, args[1] ];
```

Comment:
This now errors out, because the `local res` is missing. Either add it back, or simply replace `res` by `args` (args is not used anymore later).

The other test failure seems to be due to the changed warning. Perhaps this tidbit is helpful (it's been for me, at least):
```gap
TestDirectory("tst/testinstall", rec(rewriteToFile:=true));
```